src/targets/services/konnectorAlerts.js

```diff
@@ -0,0 +1,260 @@
+ * @param  {object} options - Options
+ * @param  {Array.<KonnectorAlert>} options.konnectorAlerts - Alerts to include in the notification
+ * @return {NotificationView} - The konnector alerts notification view
+ */
+export const buildNotificationView = (client, options) => {
+  const notifView = new KonnectorAlertView({
+    client,
+    lang,
+    data: {},
+    locales: {
+      [lang]: dictRequire(lang)
+    },
+    ...options
+  })
+  return notifView
+}
+
+const fetchRegistryInfo = async (client, konnectorSlug) => {
+  return client.stackClient.fetchJSON('GET', `/registry/${konnectorSlug}`)
+}
+
+/**
+ * Fetches triggers, filters those for which we can send a notification, and send
+ * notifications.
+ *
+ * @param  {CozyClient} client - Cozy client
+ * @return {Promise}
+ */
+export const sendTriggerNotifications = async client => {
+  const { data: triggers } = await client.query(Q('io.cozy.triggers'))
+  const konnectorTriggers = triggers.filter(trigger =>
+    isKonnectorWorker(trigger.attributes)
+  )
+  const triggerStatesDoc = await fetchTriggerStates(client)
+  const previousStates = triggerStatesDoc.triggerStates
+
+  logger('info', `${konnectorTriggers.length} konnector triggers`)
+
+  const triggerAndNotifsInfo = (await Promise.all(
+    konnectorTriggers.map(async trigger => {
+      return {
+        trigger,
+        shouldNotify: await shouldNotify(client, trigger, previousStates)
+      }
+    })
+  )).filter(({ shouldNotify }) => shouldNotify)
+
+  const konnectorSlugs = uniq(
+    triggerAndNotifsInfo.map(({ trigger }) => {
+      return trigger.message.konnector
+    })
+  )
+
+  const konnectorNamesBySlug = fromPairs(
+    await Promise.all(
+      konnectorSlugs.map(async slug => [
+        slug,
+        get(
+          await fetchRegistryInfo(client, slug),
+          'latest_version.manifest.name'
+        )
+      ])
+    )
+  )
+
+  logger('info', `${triggerAndNotifsInfo.length} konnector triggers to notify`)
+
+  const notifView = buildNotificationView(client, {
+    konnectorAlerts: triggerAndNotifsInfo.map(({ trigger }) => {
+      const konnectorSlug = trigger.message.konnector
+      return {
+        konnectorSlug,
+        konnectorName: konnectorNamesBySlug[konnectorSlug] || konnectorSlug,
+        trigger
+      }
+    })
+  })
+  await sendNotification(client, notifView)
+  await storeTriggerStates(client, triggers, triggerStatesDoc)
+}
+
+const main = async ({ client }) => {
+  client.registerPlugin(flag.plugin)
+  await client.plugins.flags.refresh()
+
+  if (!flag('banks.konnector-alerts')) {
+    logger(
+      'info',
+      'Bailing out of job notifications service since flag "banks.konnector-alerts" is not set'
+    )
+    return
+  }
+
+  logger('info', 'Executing job notifications service...')
+  await sendTriggerNotifications(client)
+}
+
+if (require.main === module || process.env.NODE_ENV === 'production') {
+  runService(main)
+}
+
+export default main
```

Comment:
maybe we could have split into a few files to lighten the main file

Reply:
I extracted the notification view. I think it's OK now.